File: app-react/src/pages/flutter/ui/tab/BottomNavigationBar.js
import React from 'react'
// ===================================================================== public component
import ListComponent from '@cpt/list.component'
// ===================================================================== component
const data = {
	title:'tab 导航',
	keyword:'BottomNavigationBar( )',
	type:'BottomNavigationBar',
	url:'BottomNavigationBarPage',
	data:[
		{ 
			keyword:'items', type:'List<BottomNavigationBarItem>', pro:'选项按钮', 
			code:[{path:'flutter/ui/tab/BottomNavigationBar/items'}],
			link:[{title:'BottomNavigationBarItem', url:'/flutter/style/BottomNavigationBarItem'}]
		},
		{ keyword:'currentIndex', type:'int', pro:'当前索引值，默认 0' },
		{ keyword:'elevation', type:'double', pro:'阴影' },
		{ keyword:'type', type:'BottomNavigationBarType', pro:'类型', code:[{path:'flutter/ui/tab/BottomNavigationBar/BottomNavigationBarType'}] },
		{ keyword:'fixedColor', type:'Color', pro:'当前tab文字颜色', link:'/flutter/style/Colors' },
		{ keyword:'backgroundColor', type:'Color', pro:'整个背景颜色', link:'/flutter/style/Colors' },
		
		{ deviceTitle:'tab color 设置', keyword:'selectedItemColor', type:'Color', pro:'当前tab文字颜色，与 fixedColor 互斥', link:'/flutter/style/Colors' },
		{ keyword:'unselectedItemColor', type:'Color', pro:'非当前tab文字颜色', link:'/flutter/style/Colors' },
		
		{ deviceTitle:'tab icon 设置', keyword:'iconSize', type:'double', pro:'icon 尺寸' },
		{ keyword:'selectedIconTheme', type:'IconThemeData', pro:'当前 icon 主题', link:'/flutter/style/IconThemeData' },
		{ keyword:'unselectedIconTheme', type:'IconThemeData', pro:'非当前 icon 主题', link:'/flutter/style/IconThemeData'  },
		
		{ deviceTitle:'tab 字体设置', keyword:'selectedFontSize', type:'double', pro:'当前tab文字大小'},
		{ keyword:'unselectedFontSize', type:'double', pro:'非当前tab文字大小'},
		
		{ deviceTitle:'label 设置', keyword:'selectedLabelStyle', type:'TextStyle', pro:'当前tab文本样式'},
		{ keyword:'unselectedLabelStyle', type:'TextStyle', pro:'非当前tab文本样式'},
		{ keyword:'showSelectedLabels', type:'bool', pro:'是否显示当前 tab 的 labels'},
		{ keyword:'showUnselectedLabels', type:'bool', pro:'是否显示非当前 tab 的 labels'},
		
		{ deviceTitle:'事件', keyword:'onTap', type:'void Function( int )', pro:'点击时执行', code:[{path:'flutter/ui/tab/onTap'}] },
	]
}
export default class Index extends React.Component{ render(){ return <ListComponent data={data} {...this.props}/> } }

File: app-flutter/lib/pages/flutter/ui/tab/BottomNavigationBar.dart
import 'package:flutter_web/material.dart';
// ======================================================================================================== 全局配置
import 'package:flutter_web_study.app/config.dart';
// ======================================================================================================== Widget 控件
class BottomNavigationBarPage extends StatefulWidget { // 有状态的控件，控件的状态、文字等属性存在变化的可能
    final String title;
    BottomNavigationBarPage({ Key key, this.title }) : super(key: key);
    @override
    _PageState createState() => _PageState(); // === createState()  => _PageState();
}
class _PageState extends State<BottomNavigationBarPage> {
    int _currentIndex = 0;
    // page
    @override
    Widget build(BuildContext context) {
        return Scaffold(
            appBar: Config.appBar(widget.title),
            body: Column(
                mainAxisAlignment:MainAxisAlignment.spaceBetween,
                children: <Widget>[
                    Expanded(
                        flex: 1,
                        child: ListView(
                            children: <Widget>[
                                Container(height: 100.0, color: Colors.yellow,child: Text('内容')),
                                Container(height: 100.0, color: Colors.blue,child: Text('内容')),
                                Container(height: 100.0, color: Colors.green,child: Text('内容')),
                                Container(height: 100.0, color: Colors.pink,child: Text('内容')),
                                Container(height: 100.0, color: Colors.yellow,child: Text('内容')),
                                Container(height: 100.0, color: Colors.blue,child: Text('内容')),
                                Container(height: 100.0, color: Colors.green,child: Text('内容')),
                                Container(height: 100.0, color: Colors.pink,child: Text('内容')),
                            ],
                        )
                    ),
                    BottomNavigationBar(
                        currentIndex: _currentIndex,
//                        fixedColor:Colors.red,
                        backgroundColor: Colors.white30,
                        selectedItemColor: Colors.red,
                        unselectedItemColor: Colors.black12,
                        selectedIconTheme:IconThemeData.fallback(),
                        selectedLabelStyle: TextStyle(color:Colors.red),
                        items: <BottomNavigationBarItem>[
                            BottomNavigationBarItem(
                                icon:  CircleAvatar(child: Icon(Icons.home, color: Colors.white),backgroundColor: Colors.black12),
                                title: Text('Home'),
                                activeIcon: CircleAvatar(child: Icon(Icons.home, color: Colors.white),backgroundColor: Colors.red,),
                            ),
                            BottomNavigationBarItem(
                                icon:  CircleAvatar(child: Icon(Icons.nature, color: Colors.white),backgroundColor: Colors.black12),
                                title: Text('News'),
                                activeIcon: CircleAvatar(child: Icon(Icons.nature, color: Colors.white),backgroundColor: Colors.red,),
                            ),
                            BottomNavigationBarItem(
                                icon:  CircleAvatar(child: Icon(Icons.center_focus_strong, color: Colors.white),backgroundColor: Colors.black12),
                                title: Text('Center'),
                                activeIcon: CircleAvatar(child: Icon(Icons.center_focus_strong, color: Colors.white),backgroundColor: Colors.red,),
                            ),
                        ],
                        showSelectedLabels:true,
                        type: BottomNavigationBarType.fixed,
                        onTap: (int index){
                            print(index);
                            setState(() {
                                _currentIndex = index;
                            });
                        },
                    ),
                ],
            )
        );
    }
}